### ISSUE TITLE
Tracking issues when trying to reused JS parser to implement json parser

### ISSUE THREAD

**Issue by IWANABETHATGUY** [rome/tools#2474]
### Description

We want to reused our JS parser to implement json parser, which could reduce a mount of work, just like `Typescript` does, https://github.com/microsoft/TypeScript/blob/main/src/compiler/parser.ts#L953-L1040

**Comment by IWANABETHATGUY** [comment]
Found a true negative case, when try to reuse js lexer, 
```rs
use rome_js_parser::lexer::Lexer;
use rome_js_parser::LexContext;
use rome_js_syntax::T;
fn main() {
    let mut lexer = Lexer::from_str(r#"{"tes": "\u{2300}"}"#, 0);
    while lexer.current() != T![EOF] {
        let a = lexer.next_token(LexContext::default());
        println!("{:?}, {:?}", a, lexer.current_range());
    }
    println!("{:#?}", lexer.finish());
}
```
actually, `"\u{2300}"` is not legally acording to json spec, 
but got the result below: 
```bash
L_CURLY, 0..1
JS_STRING_LITERAL, 1..6
COLON, 6..7
WHITESPACE, 7..8
JS_STRING_LITERAL, 8..18
R_CURLY, 18..19
EOF, 19..19
[]
```

**Comment by IWANABETHATGUY** [comment]
Closed becuase of #2785
